File: RecommendationSystem/Post.py
class Post:
    def __init__(self, post: dict):
        self.title = post["title"]
        self.content = post["content"]
        self.uuid = post["uuid"]
        self.tags = post["tags"]
        self.language = post["language"]
        self.created_at = post["created_at"]

    def to_dict(self) -> dict:
        return {
            "title": self.title,
            "content": self.content,
            "uuid": self.uuid,
            "tags": self.tags,
            "language": self.language,
            "created_at": self.created_at,
        }

    def from_dict(self, post: dict) -> None:
        self.title = post["title"]
        self.content = post["content"]
        self.uuid = post["uuid"]
        self.tags = post["tags"]
        self.language = post["language"]
        self.created_at = post["created_at"]

File: server/ExpertBridge.Api/Core/Entities/Post/Post.cs
using ExpertBridge.Api.Core.Entities.ManyToManyRelationships.PostTag;
using ExpertBridge.Api.Core.Entities.Media.PostMedia;
using ExpertBridge.Api.Core.Entities.Votes.PostVote;

namespace ExpertBridge.Api.Core.Entities.Post;

public class Post
{
    public string Id { get; set; } = Guid.NewGuid().ToString();
    public string Title { get; set; }
    public string Content { get; set; }
    public string AuthorId { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime? LastModified { get; set; }
    public bool isDeleted { get; set; }

    // Add to navigation properties
    public Profile.Profile Author { get; set; }
    public ICollection<PostMedia> Medias { get; set; } = [];
    public ICollection<Comment.Comment> Comments { get; set; } = [];
    public ICollection<PostVote> Votes { get; set; } = [];
    public ICollection<PostTag> PostTags { get; set; } = [];
}
